src/components/Tag/Tag.tsx

```diff
@@ -10,40 +10,73 @@ import styles from './Tag.scss';
 
 export interface NonMutuallyExclusiveProps {
   /** Content to display in the tag */
-  children?: string;
+  children?: string | React.ReactElement;
   /** Disables the tag  */
   disabled?: boolean;
-  /** Callback when tag is clicked or keypressed. Renders without remove button when set. */
+  /** Callback when tag is clicked or keypressed. Renders without remove button or url when set. */
   onClick?(): void;
   /** Callback when remove button is clicked or keypressed. */
   onRemove?(): void;
+  /** A string to use when tag has more than textual content */
+  title?: string;
+  /** Url to navigate to when tag is clicked or keypressed. */
+  url?: string;
```

Comment:
Let's give the `title` prop a descriptive name of `accessibilityLabel` (this also aligns with with naming in other components, like Button etc).

```suggestion
  children?: string | React.ReactElement;
  /** Disables the tag  */
  disabled?: boolean;
    /** Descriptive text to provide context for screen reader users. */
  accessibilityLabel?: string;
  /** Url to navigate to when tag is clicked or keypressed. */
  url?: string;
  /** Callback when tag is clicked or keypressed. Renders without remove button or url when set. */
  onClick?(): void;
  /** Callback when remove button is clicked or keypressed. */
  onRemove?(): void;
```